subprojects/core/src/integTest/groovy/org/gradle/api/tasks/IncrementalBuildSymlinkHandlingIntegrationTest.groovy

```diff
@@ -122,16 +122,13 @@ task work {
         result.assertTasksSkipped(":work")
     }
 
-    def "symlink may not reference missing input file"() {
+    def "symlink may reference missing input file"() {
```

Comment:
An `@InputFile` is not allowed to reference an actually missing file, so I would say it isn't allowed to reference a broken symlink. If you want to test if symlinks are allowed as inputs, I think you should use `@InputFiles`, which does not do any validation.

Reply:
Reverted